packages/plugin-team/src/decrypter.js

```diff
@@ -0,0 +1,43 @@
+/**!
+ *
+ * Copyright (c) 2015-2016 Cisco Systems, Inc. See LICENSE file.
+ * @private
+ */
+
+import {Decrypter} from '@ciscospark/plugin-conversation';
+
+Object.assign(Decrypter.prototype, {
+  /**
+   * Decrypts a team
+   * @param {Encryption~Key|string} key
+   * @param {Object} team
+   * @returns {Promise} Resolves with the decrypted team
+   * @private
+   */
+  decryptTeam(key, team) {
+    let promises = [];
+    if (team.conversations.items) {
+      promises = team.conversations.items.map((item) => this.decryptObject(item));
+    }
+
+    const usableKey = key || team.encryptionKeyUrl;
+
+    if (usableKey) {
+      promises.push(this.decryptProperty(`displayName`, usableKey, team));
+      promises.push(this.decryptProperty(`summary`, usableKey, team));
+    }
+
+    return Promise.all(promises);
+  },
+
+  /**
+   * Decrypts a Summary property
+   * @param {Encryption~Key|string} key
+   * @param {string} summary
+   * @returns {Promise} Resolves with decypted string
+   * @private
+   */
+  decryptPropSummary(key, summary) {
```

Comment:
The encrypter, decrypter, and normalizer for the team plugin are an interesting case. Because teams is basically an extension of the conversation plugin there is a lot of code that is reused. 

The way that these three files work is that they strap these methods onto the prototype of their conversation counterpart. Then, when this file gets included, these methods are added to the conversation's decrypter and then the decrypter has a decryptTeam function.

it's not perfect, and Ian and I are working on how we can make the encryption code easier to share without having to break the abstraction so much, but it's a starting point.